src/test/java/unit/WorkspaceSetSuppressLogin.java

```diff
@@ -0,0 +1,224 @@
+    super.cleanupOnce();
+
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + sharedWorkspaceId);
+
+    // `terra workspace delete --quiet`
+    TestCommand.runCommandExpectSuccess("workspace", "delete", "--quiet");
+  }
+
+  @Test
+  @DisplayName("workspace id can be set before logging in, and metadata loads after logging in")
+  void workspaceLoadsOnlyAfterLogin() throws IOException {
+    // `terra auth revoke`
+    TestCommand.runCommandExpectSuccess("auth", "revoke");
+
+    // `terra workspace set --id=$id --suppress-login`
+    UFWorkspace workspaceSet =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFWorkspace.class, "workspace", "set", "--id=" + getWorkspaceId(), "--suppress-login");
+    assertEquals(
+        getWorkspaceId(), workspaceSet.id, "workspace set before login includes workspace id");
+    assertNull(
+        workspaceSet.googleProjectId,
+        "workspace set before login does not include google project id");
+
+    // `terra status`
+    UFStatus status = TestCommand.runAndParseCommandExpectSuccess(UFStatus.class, "status");
+    assertEquals(
+        getWorkspaceId(), status.workspace.id, "status before login includes workspace id");
+    assertNull(
+        status.workspace.googleProjectId, "status before login does not include google project id");
+
+    // `terra auth status`
+    UFAuthStatus authStatus =
+        TestCommand.runAndParseCommandExpectSuccess(UFAuthStatus.class, "auth", "status");
+    assertNull(authStatus.userEmail, "auth status before login does not include user email");
+    assertNull(
+        authStatus.serviceAccountEmail, "auth status before login does not include pet SA email");
+
+    workspaceCreator.login();
+
+    // `terra status`
+    status = TestCommand.runAndParseCommandExpectSuccess(UFStatus.class, "status");
+    assertEquals(getWorkspaceId(), status.workspace.id, "status after login includes workspace id");
+    assertNotNull(
+        status.workspace.googleProjectId, "status after login includes google project id");
+
+    // `terra auth status`
+    authStatus = TestCommand.runAndParseCommandExpectSuccess(UFAuthStatus.class, "auth", "status");
+    assertNotNull(authStatus.userEmail, "auth status after login includes user email");
+    assertNotNull(authStatus.serviceAccountEmail, "auth status after login includes pet SA email");
+  }
+
+  @Test
+  @DisplayName(
+      "workspace metadata fails to load after logging in as a user without read access, then succeeds with a different workspace that they do have access to")
+  void workspaceLoadFailsWithNoAccess() throws IOException {
+    // `terra auth revoke`
+    TestCommand.runCommandExpectSuccess("auth", "revoke");
+
+    // `terra workspace set --id=$id --suppress-login`
+    TestCommand.runCommandExpectSuccess(
+        "workspace", "set", "--id=" + getWorkspaceId(), "--suppress-login");
+
```

Comment:
This comment at 113 confuses me a little; does the exception referenced get caught/thrown/ignored as part of the `workspaceSharee.login()` call, or is it something that happens in a subsequent call through TestCommand?

Reply:
Ah this comment was just out of date. There's no exception thrown, but there is a message printed to stdout that we failed to load the workspace. I decided against throwing an exception because the login should still succeed even if the user doesn't have access to the workspace that was already set. Updated comment in the test and added a check for the error message.